alibi_detect/cd/pytorch/lsdd.py

```diff
@@ -83,16 +83,20 @@ def __init__(
         else:
             self.device = torch.device('cpu')
 
+        # TODO: TBD: the several type:ignore's below are because x_ref is typed as an np.ndarray
```

Comment:
Do you think `--allow-redefinitions` would cause mypy to miss anything significant elsewhere? I wonder if it's best to activate it now as redefining variable types is pretty common for us? 

Otherwise, we could go with a convention of renaming variables instead, i.e. something like
```
x_ref_torch = cast(torch.Tensor, torch.as_tensor(self.x_ref).to(self.device)
```

Although the latter seems undesirable to me...

Reply:
Renaming wouldn't need the `cast` at all I think.

I'm a bit wary of enabling `--allow-redefinitions` as it's unclear as you say what we could potentially miss. Even so, it is quite limited as redefinitions would be allowed only within the same scope.

Reply:
Sorry yes `cast` is making more sense to me now!

I'd have thought that within the same scope would mostly be fine for us? But in any case, it's not clear to me what "within the same block and nesting depth" in the mypy docs means. Also, I take your point that it's probably worth being cautious about changing this.

Reply:
Well I don't know what happened but I just removed all these ignore's and locally the `mypy` check passes now...

Reply:
Nevermind, I was being silly as had downgraded back to `numpy 1.19`...

Reply:
Too good to be true!